### PR TITLE
Adding 1.4.0 operator yaml

### PR DESCRIPTION
**Pull request by rajdas98**
Signed-off-by: Raj <raj.das@mayadata.io>
- Adding 1.4.0 operator yaml

### CODE REVIEW

File: docs/litmus-operator-v1.4.0.yaml [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+---
+# Source: openebs/templates/clusterrole.yaml
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: litmus
+  labels:
+    name: litmus
+rules:
+  - apiGroups: ["","apps","batch","litmuschaos.io","openshift.io"]
```
Comment (rahulchheda) [inline docs/litmus-operator-v1.4.0.yaml:22]: - apiGroups: ["admissionregistration.k8s.io"]
  resources: ["validatingwebhookconfigurations"]
  verbs: ["get", "create", "list", "delete", "update"]

File: docs/litmus-operator-v1.4.0.yaml [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    resources: ["pods","jobs","deployments","replicationcontrollers","daemonsets","replicasets","deploymentconfigs","statefulsets","events","configmaps","services","chaosengines","chaosexperiments","chaosresults"]
```
Comment (rahulchheda) [inline docs/litmus-operator-v1.4.0.yaml:23]: add `"secrets"`

File: docs/litmus-operator-v1.4.0.yaml (hunk 1 above)
Comment (rajdas98) [reply docs/litmus-operator-v1.4.0.yaml:22]: done

File: docs/litmus-operator-v1.4.0.yaml [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+# Source: openebs/templates/clusterrole.yaml
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: litmus
+  labels:
+    name: litmus
+rules:
+  - apiGroups: ["","apps","batch","litmuschaos.io","openshift.io"]
+    resources: ["pods","jobs","deployments","replicationcontrollers","daemonsets","replicasets","deploymentconfigs","statefulsets","events","configmaps","services","secrets","chaosengines","chaosexperiments","chaosresults"]
```
Comment (rahulchheda) [inline docs/litmus-operator-v1.4.0.yaml:23]: Can you add "events" also, we can give this RBAC to event-router as well.

File: docs/litmus-operator-v1.4.0-RC1.yaml [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: litmus
+  namespace: litmus
+  labels:
+    name: litmus
+---
+# Source: openebs/templates/clusterrole.yaml
```
Comment (rahulchheda) [inline docs/litmus-operator-v1.4.0-RC1.yaml:14]:
```suggestion

```

File: docs/litmus-operator-v1.4.0-RC1.yaml [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+      name: chaos-operator
+  template:
+    metadata:
+      labels:
+        name: chaos-operator
+    spec:
+      serviceAccountName: litmus
+      containers:
+        - name: chaos-operator
+          image: "litmuschaos/chaos-operator:1.4.0-RC1"
```
Comment (rahulchheda) [inline docs/litmus-operator-v1.4.0-RC1.yaml:62]:
```suggestion
          image: "litmuschaos/chaos-operator:1.4.0"
```

File: docs/litmus-operator-v1.4.0-RC1.yaml [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+          command:
+            - chaos-operator
+          imagePullPolicy: Always
+          env:
+            - name: CHAOS_RUNNER_IMAGE
+              value: "litmuschaos/chaos-runner:1.4.0-RC1"
```
Comment (rahulchheda) [inline docs/litmus-operator-v1.4.0-RC1.yaml:68]:
```suggestion
              value: "litmuschaos/chaos-runner:1.4.0"
```

**Comment by rahulchheda** [comment]
Please rename the file to `docs/litmus-operator-v1.4.0.yaml `

**Comment by rahulchheda** [comment]
Please update the CRD validation

File: docs/litmus-operator-v1.4.0.yaml [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+              type: string
+              pattern: ^(active|stop|initialized|stopped)$
+            chaosServiceAccount:
+              type: string
+            components:
+              type: object
+              properties:
+                runner:
+                  type: object
+                  properties:
```
Comment (ksatchit) [inline docs/litmus-operator-v1.4.0.yaml:143]: runnerAnnotation changes?

File: docs/litmus-operator-v1.4.0.yaml [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+              items:
+                type: object
+                properties:
+                  name:
+                    type: string
+                  spec:
+                    type: object
+                    properties:
+                      components:
+                        type: object
```
Comment (ksatchit) [inline docs/litmus-operator-v1.4.0.yaml:160]: experimentAnnotation changes?

File: docs/litmus-operator-v1.4.0.yaml (hunk 7 above)
Comment (rajdas98) [reply docs/litmus-operator-v1.4.0.yaml:142]: Nope, compared with 1.3

File: docs/litmus-operator-v1.4.0.yaml (hunk 8 above)
Comment (rajdas98) [reply docs/litmus-operator-v1.4.0.yaml:159]: Nope, compared with 1.3

File: docs/litmus-operator-v1.4.0.yaml (hunk 7 above)
Comment (rajdas98) [reply docs/litmus-operator-v1.4.0.yaml:142]: Sorry, you right

**Pull request merged**